src/table/utils/styling-utils.js

```diff
@@ -150,6 +150,20 @@ export function getBodyCellStyle(layout, theme) {
   };
 }
 
+const isTotalModeAuto = (layout) => layout.totals?.show === true;
+const hasOnlyMeasure = (layout) => layout.qHyperCube.qDimensionInfo.length === 0;
+const hasTotalsMeasures = (layout) => {
+  return layout.qHyperCube.qMeasureInfo.length > 0 && layout.qHyperCube.qGrandTotalRow.length > 0;
+};
+
+export function showTotals(layout) {
+  if (isTotalModeAuto(layout) && hasOnlyMeasure(layout)) return false;
+  return hasTotalsMeasures(layout) && layout.totals && (layout.totals.show || layout.totals.position !== 'noTotals');
+}
+
+export function showTotalsAtTop(layout) {
+  return !(showTotals(layout) && layout.totals && !layout.totals.show && layout.totals.position === 'bottom');
+}
```

Comment:
This doesn't really relate to styling. I think it would make sense to be a seperate file.